core/common-util/src/test/java/uk/gov/gchq/gaffer/commonutil/iterable/LimitedInMemorySortedIterableTest.java

```diff
@@ -90,6 +90,19 @@ void shouldLimitAndNotDeduplicateEntries() {
         assertThat(list).containsExactly(1, 1, 2, 2);
     }
 
+    @Test
+    void shouldInsertNewItemAndRemoveOldItemWhenlimitIsReachedAndNewItemIsLessThanLastEntry() {
+        final LimitedInMemorySortedIterable<Integer> list = new LimitedInMemorySortedIterable<Integer>(Comparator.naturalOrder(), 3, false);
+
+        list.add(3);
+        list.add(2);
+        list.add(3);
+        list.add(2);
+
+        assertThat(list.size()).isEqualTo(3);
```

Comment:
Could chain the assert to be `assertThat(list).hasSize(3).containsExactly(2, 2, 3)`